File: node/examples/01-chat-logger.js
// Log every chat message from a TikTok LIVE stream.
// Run: USERNAME=streamername node 01-chat-logger.js
import { TikTokLive } from 'tiktok-live-events';

const live = new TikTokLive(process.env.USERNAME || 'tiktokuser');

live.on('connected', () => console.log('[live] connected'));
live.on('chat', e => console.log(`${e.user.uniqueId}: ${e.comment}`));

await live.connect();

File: python/examples/01_chat_logger.py
"""Log every chat message from a TikTok LIVE stream.

Run:
    USERNAME=streamername python 01_chat_logger.py
"""
import asyncio
import os
from tiktok_live_events import TikTokLive

live = TikTokLive(os.environ.get("USERNAME", "tiktokuser"))


@live.on("connected")
def on_connected(_):
    print("[live] connected")


@live.on("chat")
def on_chat(e):
    print(f"{e['user']['uniqueId']}: {e['comment']}")


asyncio.run(live.run())
